src/components/document/document-workspace.tsx

```diff
@@ -207,6 +210,30 @@ export class DocumentWorkspaceComponent extends BaseComponent<IProps, {}> {
     }
   }
 
+  private handleDeleteDocument = (document: DocumentModelType) => {
+    this.stores.ui.confirm(`Delete this workspace?`, `Delete Workspace`)
+      .then((deleted: boolean) => {
+        const docType = document.type;
+        const invalidDocs = ProblemDocument || PublicationDocument || PersonalPublication ||
+                            LearningLogPublication || SupportPublication;
+        if (docType !== invalidDocs) {
+          document.setProperty("softDelete", "true");
```

Comment:
This appears to delete the document even if the user cancels the delete. Presumably, neither the delete nor the document switch should happen if the user cancels.

Separately, I think I'd be inclined to call the property `isDeleted`. From the user's perspective it's been deleted and the fact that we don't actually get rid of it is an implementation detail.